pkg/operator/k8sutil/network.go

```diff
@@ -80,13 +81,22 @@ func ApplyMultus(net cephv1.NetworkSpec, objectMeta *metav1.ObjectMeta) error {
 			shortSyntax = true
 		}
 
-		var isExcluded bool
-		for _, clusterNetworkApps := range getClusterNetworkApps() {
-			isExcluded = strings.Contains(objectMeta.Labels["app"], clusterNetworkApps)
+		app, ok := objectMeta.Labels["app"]
+		if !ok {
+			app = "" // unknown app
 		}
-		if isExcluded {
+		isClusterNetApp := false
+		for _, clusterNetworkApp := range getClusterNetworkApps() {
+			if app == clusterNetworkApp {
```

Comment:
This could be changing the behavior since the label must be exactly `app=rook-ceph-osd` instead of all pods containing `osd`. What about the prepare jobs? I'm not sure if they need this, but until it's been tested in a multus environment we may not want to change it.

Reply:
Given this question, I think it might be good to do the cleanup in another PR. 

I think it is for the best if we clearly define the apps that should get cluster network info. 

AFAICT from looking at code paths, the OSDs only `ApplyMultus()` from `makeDeployment()` which is only used for OSDs on PVC/Node and not for provisioning (which uses `provisionPodTemplateSpec()`). I'm pretty sure this means only `rook-ceph-osd` pods apply.